src/hooks/transactions/batch/tracker/useUpdateBatch.ts

```diff
@@ -81,6 +86,28 @@ export function useUpdateBatch() {
             })
           );
         }
+      } else {
```

Comment:
there seem to be many nested statements here. You could consider some small function to handle this logic